pkg/api/api.go

```diff
@@ -353,25 +348,18 @@ func (s *Service) Close() error {
 
 // getOrCreateTag attempts to get the tag if an id is supplied, and returns an error if it does not exist.
 // If no id is supplied, it will attempt to create a new tag with a generated name and return it.
-func (s *Service) getOrCreateTag(tagUid string) (*tags.Tag, bool, error) {
+func (s *Service) getOrCreateSessionID(tagUid uint64) (uint64, error) {
```

Comment:
You can register this as a hook in `preMapHooks` similar to the `resolveNameOrAddress` method. That way you'll just use the tag on the struct.

Reply:
So the problem is the usage is different in different APIs. For eg we create a tag only when deferred upload/pin flag is set in header. Can this be handled in premaphook?